### PR TITLE
new: append to base path if one exists

### LINKED ISSUE #75: Client URL x Request Path
More of a **question** than an issue.

When I create a JsonClient in node I do the following:

``` js
var client = restify.createJsonClient({
  url: 'https://www.domain.com:4321/api'
});
```

Once I've done that, I make calls like so:

``` js
client.post('/service/path', { });
```

Which seems right. I expect that the path called would be something like `https://www.domain.com:4321/api/service/path`. However, what is happening is that the client is throwing away the /api base path and calling `https://www.domain.com:4321/service/path`.

I don't get it - I'm inserting the client URL into a config file, so that I can change hosts without any hassle; Now that I need a base path, I need to change the code as well as the config.

### PR DESCRIPTION
**Pull request by DonutEspresso**
Fixes #75 

It's unclear that the existing code ever worked as intended since `this.path` was never set. It turns out that fixing this behavior is a breaking change, so I gated it behind an option.

### CODE REVIEW

File: lib/HttpClient.js [hunk 1]
```diff
@@ -484,8 +485,10 @@ function HttpClient(options) {
             'must specify http/https protocol!'
         );
         this.url = parsedUrl;
+        this.path = parsedUrl.pathname;
```
Comment (misterdjules) [inline lib/HttpClient.js]: Since we're using `pathname` and not `path`, what happens if users call:
```
var client = restify.createJsonClient({
  url: 'http://foo.com:42/bar?foo=bar' // note the query string here
});

client.get('/foo', ...)
```
?

Is that considered invalid? If so should it throw an error? Or do we consider that it's fine to perform the request by dropping the query string?

Comment (DonutEspresso) [reply lib/HttpClient.js]: Good observation! Query strings are already silently thrown away (prior to this PR), with the preferred way of doing this to set a `query` option in the constructor. We don't enforce this today so while we probably should throw, I'd reserve it for a breaking change. 

In general, the idea is that the constructor options are the "default" values, and those can be overridden/stomped at verb time (`get()` or `post()`) if you pass in an option with the same name. There's some clean up we need to do to actually ensure that this is consistent for all options (as identified in this PR), but also some questions about how (or if) we want to support merging vs stomping of options - I imagine it might very well be a use case to want to always merge with default `query` instead of stomping. 

With the formalization of the URL class (https://nodejs.org/api/url.html#url_class_url) we should probably make a clean break in a new major version and jettison inconsistent and legacy options names where possible.

Comment (DonutEspresso) [reply lib/HttpClient.js]: Yep! It is saved to `.url`, which is then passed to raw `http|https.request()` here:
https://github.com/restify/clients/blob/master/lib/HttpClient.js#L190

However, it seems the query value which was parsed, `?foo=bar` isn't used by the http module. As far as I can tell in local testing it's thrown away. I can add a test to actually verify that behavior hasn't changed (and can revisit in said breaking change).

File: .travis.yml [hunk 2]
```diff
@@ -4,9 +4,7 @@ script:
 node_js:
     - "4"
     - "6"
-    - "7"
```
Comment (hekike) [inline .travis.yml]: Maybe we should follow the same approach:
https://github.com/restify/node-restify/blob/master/.travis.yml#L4

Comment (DonutEspresso) [reply .travis.yml]: We can, but stable (node10) is currently broken, even in restify. Something changed in latest 10.x that broke us - need to look into it.

File: lib/HttpClient.js [hunk 3]
```diff
@@ -443,6 +443,7 @@ function HttpClient(options) {
     var self = this;
 
     this.agent = options.agent;
+    this.appendPath = options.appendPath || false;
```
[the remark this answers is not in the document]
Comment (DonutEspresso) [reply lib/HttpClient.js]: Added!

File: test/e2e.js [hunk 4]
```diff
@@ -8,8 +8,9 @@ var clients = require('../lib');
 
 describe('restify-client tests against real web server', function () {
     it('have timings', function (done) {
-        var client = clients.createJsonClient({
-            url: 'https://netflix.com'
+        this.timeout(10000);
+        var client = clients.createStringClient({
```
Comment (DonutEspresso) [inline test/e2e.js]: This is a somewhat unrelated change - just upping the timeout here and using String client instead of JSON.

**Comment by DonutEspresso** [comment]
@misterdjules once more? Magically, this PR didn't break any of the tests we added in the other PR (!!).

**Comment by DonutEspresso** [comment]
FYI, node 10 is broken (known issue), so Travis will continue to fail for now.
